From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Richard Townsend <richard.townsend@arm.com>
Date: Mon, 3 Jun 2019 09:52:49 +0100
Subject: build: pull in a fixed compiler for Windows on Arm

Due to a code-generation defect in the version of Clang used for the M76
branch related to virtual method thunks, it's necessary to build M76
with a later version of Clang. This change pulls in a corrected version
by setting ELECTRON_BUILDING_WOA=1 or similar in the environment.

This PR is only intended to be a temporary workaround and will be
removed when Electron's Chromium updates to a compiler unaffected by
this issue.

diff --git a/tools/clang/scripts/update.py b/tools/clang/scripts/update.py
index 12831cfa89d6dba67f49628b8f2cb8f3d9ac97f0..6ff434b1406de3e809218d4987e8ea0b4a647e43 100755
--- a/tools/clang/scripts/update.py
+++ b/tools/clang/scripts/update.py
@@ -41,6 +41,11 @@ CLANG_REVISION = '49b965079b18f8aa485dd1156dd088d40b7ee465'
 CLANG_SVN_REVISION = '363450'
 CLANG_SUB_REVISION = 1
 
+if os.getenv('ELECTRON_BUILDING_WOA'):
+    CLANG_REVISION = '56bee1a90a71876cb5067b108bf5715fa1c4e843'
+    CLANG_SVN_REVISION = '361657'
+    CLANG_SUB_REVISION = 1
+
 PACKAGE_VERSION = '%s-%s-%s' % (CLANG_SVN_REVISION, CLANG_REVISION[:8],
                                 CLANG_SUB_REVISION)
 RELEASE_VERSION = '9.0.0'
